src/Faker/Provider/hy_AM/PhoneNumber.php

```diff
@@ -4,34 +4,30 @@
 
 class PhoneNumber extends \Faker\Provider\PhoneNumber
 {
+
+    protected static $codes = array(91, 96, 99, 43, 77, 93, 94, 98, 97, 77, 55, 95, 41, 49);
+
+    protected static $numberFormats = array(
+        '######',
+        '##-##-##',
+        '###-###',
+    );
+
     protected static $formats = array(
-        '093 ######',
-        '093 ##-##-##',
-        '(093) ######',
-        '(093) ##-##-##',
-        '094 ######',
-        '094 ##-##-##',
-        '(094) ######',
-        '(094) ##-##-##',
-        '095 ######',
-        '095 ##-##-##',
-        '(095) ######',
-        '(095) ##-##-##',
-        '096 ######',
-        '096 ##-##-##',
-        '(096) ######',
-        '(096) ##-##-##',
-        '099 ######',
-        '099 ##-##-##',
-        '(099) ######',
-        '(099) ##-##-##',
-        '077 ######',
-        '077 ##-##-##',
-        '(077) ######',
-        '(077) ##-##-##',
-        '055 ######',
-        '055 ##-##-##',
-        '(055) ######',
-        '(055) ##-##-##',
+        '0{code} {numberFormat}',
+        '(0{code}) {numberFormat}',
+        '+374{code} {numberFormat}',
+        '+374 {code} {numberFormat}',
     );
+
+    public function phoneNumber()
+    {
+        $format = static::randomElement(static::$formats);
+        $code = static::randomElement(static::$codes);
+        $numberFormat = static::randomElement(static::$numberFormats);
+
+        $number = str_replace(array('{code}', '{numberFormat}'), array($code, $numberFormat), $format);
```

Comment:
You should not need to `str_replace`, `parse()` would do it for you if there were related formatters.